crates/common/src/prov/model/mod.rs

```diff
@@ -1163,29 +1164,115 @@ impl ProvModel {
 }
 
 custom_error::custom_error! {pub CompactionError
-    JsonLd{inner: String}                  = "JsonLd {inner}", //TODO: contribute Send to the upstream JsonLD error type
-    Join{source : JoinError}               = "Tokio",
-    Serde{source: serde_json::Error }      = "Serde conversion {source}",
+    JsonLd{inner: String}                  = "JSON-LD: {inner}", //TODO: contribute Send to the upstream JsonLD error type
+    Join{source : JoinError}               = "Tokio: {source}",
+    Serde{source: serde_json::Error}       = "Serde conversion: {source}",
+    InvalidExpanded{message: String}       = "Expanded document invalid: {message}",
+    NoObject{document: Value}              = "Compacted document not a JSON object: {document}",
+}
+pub struct ExpandedJson(pub serde_json::Value);
```

Comment:
This seems a bit unwieldy, is there no simple way to go from a context in json format to a ..context?

Reply:
If about `construct_context_definition`: I'd definitely expected one but I couldn't find it!

Reply:
Now wrapped it in a `lazy_static` at least though so we need do it no more than once.